src/main/java/com/google/cloud/genomics/utils/grpc/MergeAllVariantsAtSameSite.java

```diff
@@ -55,78 +57,146 @@ public void merge(Long windowStart, Iterable<Variant> variants, VariantEmitterSt
+    noteOverlappingVariantCalls(currentStartVariantRecords, priorStartVariantRecords);
+    mergeOverlappingBlockRecordCalls(currentStartVariantRecords, blockRecords);
+
+    // Emit them to the underlying processing engine.
+    for (Variant.Builder variantBuilder : currentStartVariantRecords) {
+      Variant variant = variantBuilder.build();
+      emitter.emit(variant);
+    }
+
+    // Reset our state.
+    currentStartVariantRecords.clear();
+    priorStartVariantRecords.addAll(copyOfCurrentStartVariantRecords);
+  }
+
+  static void mergeOverlappingBlockRecordCalls(List<Variant.Builder> currentStartVariantRecords,
+      List<Variant> blockRecords) {
+
+    // Identify the block records overlapping this position.
+    List<VariantCall> overlappingBlockRecordCalls = new ArrayList();
+    for (Iterator<Variant> iterator = blockRecords.iterator(); iterator.hasNext();) {
+      Variant blockRecord = iterator.next();
+      if (VariantUtils.isOverlapping(blockRecord, currentStartVariantRecords.get(0))) {
+        overlappingBlockRecordCalls.addAll(blockRecord.getCallsList());
+      } else if (blockRecord.getStart() < currentStartVariantRecords.get(0).getStart()) {
+        // Remove the current element from the iterator and the list since it is
+        // left of this position.
+        iterator.remove();
+      } else {
+        // Break out of this loop since we are now examining records to the right of this position.
+        break;
+      }
+    }
+
+    // Add the overlapping block record calls to all variants at this position.
+    for (Variant.Builder destVariant : currentStartVariantRecords) {
+      destVariant.addAllCalls(overlappingBlockRecordCalls);
+    }
+  }
+
+  static final Function<VariantCall, Value> GET_CALL_SET_NAME = new Function<VariantCall, Value>() {
+    @Override
+    public Value apply(VariantCall call) {
+      return Value.newBuilder().setStringValue(call.getCallSetName()).build();
+    }
+  };
+
+  static void noteOverlappingVariantCalls(List<Variant.Builder> currentStartVariantRecords,
+      List<Variant> priorStartVariantRecords) {
+
+    // Identify the prior variant records overlapping this position.
+    ListValue.Builder priorCallSetNamesBuilder = ListValue.newBuilder();
+    for (Iterator<Variant> iterator = priorStartVariantRecords.iterator(); iterator.hasNext();) {
+      Variant variantRecord = iterator.next();
+      if (VariantUtils.isOverlapping(variantRecord, currentStartVariantRecords.get(0))) {
+        priorCallSetNamesBuilder.addAllValues(Lists.transform(variantRecord.getCallsList(),
+            GET_CALL_SET_NAME));
+      } else if (variantRecord.getStart() < currentStartVariantRecords.get(0).getStart()) {
+        // Remove the current element from the iterator and the list since it is
+        // left of this position.
+        iterator.remove();
+      } else {
+        // Break out of this loop since we are now examining records to the right of this position.
+        break;
```

Comment:
Here's where the sort occurs: https://github.com/deflaux/utils-java/blob/4ada4298122adddc222fd3a79de831ed2b8c0a5c/src/main/java/com/google/cloud/genomics/utils/grpc/MergeAllVariantsAtSameSite.java#L58